rcldotnet_tests/MessagePubSubTest.cs

```diff
@@ -0,0 +1,317 @@
+        //     // var multiArrayLayout = new std_msgs.msg.MultiArrayLayout();
+        //     // multiArrayLayout.Dim = multiArrayDimension;
+
+        //     var msg = new std_msgs.msg.Float64MultiArray();
+        //     // msg.Layout = multiArrayLayout;
+        //     // msg.Data = new double[2];
+
+        //     publisher.Publish(msg);
+
+        //     Rclcs.SpinOnce(sub_node, sub_context, 0.5);
+
+        //     Assert.That(callbackTriggered, Is.True);
+
+        //     publisher.Dispose();
+        //     subscription.Dispose();
+        // }
+
+        // [Test]
+        // public void ImageSub()
+        // {
+        //     bool callbackTriggered = false;
+        //     var subscription = sub_node.CreateSubscription<sensor_msgs.msg.Image>(
+        //       "/image", (received_msg) =>
+        //       {
+        //         callbackTriggered = true;
+        //         Assert.That(received_msg.Data[0], Is.EqualTo(5));
+        //       });
+
+        //     Rclcs.SpinOnce(sub_node, sub_context, 3.0);
+
+        //     Assert.That(callbackTriggered, Is.True);
+
+        //     subscription.Dispose();
+        // }
+
+        // TODO(sam): implement byte type in generators
+        // [Test]
+        // public void ByteMultiArrayPubSub()
+        // {
+        //     bool callbackTriggered = false;
+        //     var subscription = sub_node.CreateSubscription<std_msgs.msg.ByteMultiArray>(
+        //       "test_topic", (received_msg) => { callbackTriggered = true; });
+        //
+        //     var publisher = pub_node.CreatePublisher<std_msgs.msg.ByteMultiArray>("test_topic");
+        //     var msg = new std_msgs.msg.ByteMultiArray();
+        //     publisher.Publish(msg);
+        //
+        //     Rclcs.SpinOnce(sub_node, sub_context, 0.5);
+        //
+        //     Assert.That(callbackTriggered, Is.True);
+        //
+        //     publisher.Dispose();
+        //     subscription.Dispose();
+        // }
+
+        // [Test]
+        // public void EmptyUnboundedSequenses()
+        // {
+        //     bool callbackTriggered = false;
+        //     var subscription = node.CreateSubscription<test_msgs.msg.UnboundedSequences>(
+        //       "test_topic", (received_msg) => { callbackTriggered = true; });
+        //
+        //     var publisher = node.CreatePublisher<test_msgs.msg.UnboundedSequences>("test_topic");
+        //     var msg = new test_msgs.msg.UnboundedSequences();
+        //
+        //     msg.Basic_types_values = new test_msgs.msg.BasicTypes[0];
+        //     msg.Constants_values = new test_msgs.msg.Constants[0];
+        //     msg.Defaults_values = new test_msgs.msg.Defaults[0];
+        //
+        //     publisher.Publish(msg);
+        //     Rclcs.SpinOnce(node, context, 0.1);
+        //
+        //     Assert.That(callbackTriggered, Is.True);
+        //
+        //     publisher.Dispose();
+        //     subscription.Dispose();
+        // }
+
```

Comment:
What is the status of commented tests?

Reply:
These tests are a bit slow to run and should not be necessary if the message read/write tests are written properly. Had not decided if I should keep them all. Deleted all but one for now.